### PR TITLE
Refactor Button outline -> mixin and SageLabel--interactive

### PR DESCRIPTION
**Pull request by AndrwM**
- Refactors button outline logic to a mixin ( `sage-focus-outline($color)` )
- SageLabel--interactive

@philschanely and I ran through this solution and we both feel comfortable with refactoring the Button outline logic into a mixin.

Comps: https://www.figma.com/file/TO8XSqt6tgscFcShhkWVdm/Products-Admin?node-id=615%3A5427

screencast 2020-09-22 12-31-48

### CODE REVIEW

File: app/views/examples/elements/label/_preview.html.erb [hunk 1]
```diff
@@ -42,3 +42,25 @@ colors = [
...
+  Published
+</a>
+
+<a href="#" class="sage-label sage-label--interactive sage-label--warning sage-label--icon-right-caret-down">
+  Warning
+</a>
+
+<a href="#" class="sage-label sage-label--interactive sage-label--danger sage-label--icon-right-caret-down">
+  Locked
+</a>
```
Comment (AndrwM) [inline app/views/examples/elements/label/_preview.html.erb:66]: I have a followup PR that will convert this preview file to SageRails. Wanted to keep this interactive modifier PR focused.

File: lib/sage-frontend/stylesheets/system/core/_mixins.scss [hunk 2]
```diff
@@ -182,6 +182,59 @@
...
+@mixin sage-focus-outline($outline-color) {
+  position: relative;
+
+  &:focus {
+    outline: none;
+  }
+
+  &::after {
+    content: "";
+    display: inline-block;
```
Comment (philschanely) [inline lib/sage-frontend/stylesheets/system/core/_mixins.scss]: This may be redundant given the positioning configurations.

**Review by philschanely — COMMENTED** [review]
🙌 Really nice work here @andrew! One small suggestion.

**Review by philschanely — APPROVED** [review]

File: lib/sage-frontend/stylesheets/system/core/_mixins.scss (hunk 2 above)
Comment (AndrwM) [reply lib/sage-frontend/stylesheets/system/core/_mixins.scss]: I think you might be right, will remove :)

**Pull request merged**